packages/bsky/tests/views/author-feed.test.ts

```diff
@@ -232,4 +234,42 @@ describe('pds author feed views', () => {
       },
     )
   })
+
+  it('can filter by type', async () => {
+    const { data: allFeed } = await agent.api.app.bsky.feed.getAuthorFeed({
+      actor: carol,
+    })
+
+    expect(
+      allFeed.feed.some(({ post }) => {
+        return isRecord(post.record) && Boolean(post.record.reply)
+      }),
+    ).toBeTruthy()
+    expect(
+      allFeed.feed.some(({ post }) => {
+        return isEmbedRecordWithMedia(post.record)
+      }),
+    ).toBeTruthy()
+
+    const { data: mediaFeed } = await agent.api.app.bsky.feed.getAuthorFeed({
+      actor: carol,
+      filter: 'posts_with_media',
+    })
+
+    expect(
+      mediaFeed.feed.every(({ post }) => {
+        return isEmbedRecordWithMedia(post.record)
```

Comment:
this doesn't seem quite right & I'm honestly not sure why it's not throwing 🤔 

when I try to do the same checkout outside of the `.every` it throws an error

Reply:
I think you want to import `isMain` instead of `validateMain` from both embed/images & embed/recordWithMedia & do 

`return isEmbedImages(post.record.embed) || isEmbedRecordWithMedia(post.record.embed)`

Reply:
Oh, false positive, was just returning an error, not throwing, good catch